docs/content/dagster-cloud/deployment/serverless.mdx

```diff
@@ -94,41 +94,86 @@ Any dependencies specified in either `requirements.txt` or `setup.py` will be in
 
-In the root of your repo, you can provide two optional shell scripts: `dagster_cloud_pre_install.sh` and `dagster_cloud_post_install.sh`. These will run before and after Python dependencies are installed. They are useful for installing any non-Python dependencies or otherwise configuring your environment.
+### Using a different Python version
 
-### Changing the base image
+The default version of Python for Serverless deployments is Python 3.8. Versions 3.7, 3.9 and 3.10 are also supported. You can specify the version you want by updating your GitHub workflow or using the `--python-version` command line argument:
 
-This method is the most flexible, but requires setting up a pipeline outside of Dagster to build a custom base image.
+- **With GitHub**: Change the `python_version` specified in your `.github/workflows` files. For example, change [this line](https://github.com/dagster-io/quickstart-etl/blob/e07e944c7504a52b3d252553d51ad2085b4d5914/.github/workflows/deploy.yml#L44) in your `deploy.yml` to the version you want.
 
-The default base image is `debian:bullseye-slim`, but it can be changed.
+- **With the CLI**: Add the `--python-version` CLI argument to the deploy command to specify the registry path to the desired base image:
 
-- **With GitHub**: Provide a `base_image` input parameter to the **Build and deploy** step in your GitHub Actions configuration (usually at `.github/workflows/deploy.yml`):
+  ```shell
+  dagster-cloud serverless deploy --location-name=my_location --python-version=3.9
+  ```
+
+### Using a different OS or using native dependencies
+
+Your code is run on a Docker image built using:
+1. The standard Python "slim" [Docker image](https://hub.docker.com/_/python), such as `python:3.8-slim`.
+2. The `dagster-cloud[serverless]` module installed in the image.
+
+As far as possible, add all dependencies by including the corresponding native Python bindings in your `setup.py`.   When that is not possible, you can build and upload a custom base image that will be used to run your Python code. 
+
+To build and upload the image, use the command line:
+
+1. Build your docker image. Ensure you have included the `dagster-cloud[serverless]` dependency. You can do this by adding the following to your `Dockerfile`: `RUN pip install "dagster-cloud[serverless]"`.
+
+2. Get the credentials to access the Dagster cloud image registry. The following should print out the registry URL:
+
+```shell
+$ set -a
+$ eval "$(dagster-cloud serverless registry-info)"
+$ set +a
+$ echo $REGISTRY_URL
+657821118200.dkr.ecr.us-west-2.amazonaws.com/serverless-agent-UUID
+```
+
+3. Login to docker using the retrieved credentials:
+
+```shell
+$ echo $AWS_ECR_PASSWORD | docker login --username AWS --password-stdin $REGISTRY_URL
+Login Succeeded
+```
+
+3. Tag your built docker image using the `$REGISTRY_URL`:
+
+```shell
+$ docker tag build-image:current-tag $REGISTRY_URL:published-tag
```

Comment:
I had sort of been imagining that this would be exposed by some CLI rather than having users interact directly with our ECR repo (I know this is what the github action does under the hood, but putting it behind a CLI gives us a bit more flexibility to adjust it in the future I think?)

added @Ramshackle-Jamathon and @prha in case they have thoughts  here

Reply:
yeah, I like having this layer of indirection so that we can change it under the hood if we need to.

Reply:
Agreed. I'm working on a command to do this.

Reply:
What about this flow, before I implement it:

1. build your docker image as `your-image:your-tag`
2. `dagster-cloud serverless deploy-base-image your-image:your-tag published-tag`
3. Options:
    * (CLI) `dagster-cloud serverless deploy-python-executable ... --base-image-tag=published-tag`
    * (Github) `SERVERLESS_BASE_IMAGE_TAG: "published-tag"`

Reply:
This is now being changed in an internal PR. I'll revise the docs shortly.

Reply:
Done.